nitro-protocol/test/contracts/ForceMove/challenge.test.ts

```diff
@@ -255,14 +257,17 @@ describe('challenge', () => {
 });
 
 describe('challenge with transaction generator', () => {
+  const twoPartyFixedPart = {...twoPartyChannel, appDefinition, challengeDuration};
+  
   beforeEach(async () => {
-    await (await ForceMove.setStatus(getChannelId(twoPartyChannel), HashZero)).wait();
+    await (await ForceMove.setStatus(getChannelId(twoPartyFixedPart), HashZero)).wait();
   });
+  // FIX: even if dropping channel status before each test, turn nums from prev tests are saved and can cause reverts
   it.each`
     description                                     | appData   | outcome                            | turnNums  | challenger
     ${'challenge(0,1) accepted'}                    | ${[0, 0]} | ${[]}                              | ${[0, 1]} | ${1}
     ${'challenge(1,2) accepted'}                    | ${[0, 0]} | ${[]}                              | ${[1, 2]} | ${0}
-    ${'challenge(1,2) accepted, MAX_OUTCOME_ITEMS'} | ${[0, 0]} | ${largeOutcome(MAX_OUTCOME_ITEMS)} | ${[1, 2]} | ${0}
+    ${'challenge(3,4) accepted, MAX_OUTCOME_ITEMS'} | ${[0, 1]} | ${largeOutcome(MAX_OUTCOME_ITEMS)} | ${[3, 4]} | ${0}
```

Comment:
I'd like to understand this a little better. What's going wrong?

Reply:
The last test fails due to transaction revert with reason `turnNumRecord not increased`, which corresponds to `CountingApp.sol`. Both changing `turnNums` to `[3, 4]` and removing middle test help to avoid this revert. Unfortunately, I was not able to debug `CountingApp.sol` (via `console.log`) behavior with this tests because `ganache` is created dynamically and even setting env args to enable logging does not help.